### ISSUE TITLE
Why make it so complex ?

### ISSUE THREAD

**Issue by r4nchy** [vrtmrz/obsidian-livesync#396]
Why can't be have a simple livesync ?
The current state of obsidian-livesync plugin is so complicated to use and manage.

Even if it provides a lot of customization, the lack of any documentation or any sort of tutorials on how to sync multiple devices etc is really making me irk.

Why not make it as simple as  possible so you don't have to customize the crap out of everything on every device. 

Please make documentation or videos or something so we can understand how to use this thing.

**Comment by lushangkan** [comment]
If you want to use it out of the box, please use the official Sync service. Also, there is already documentation for this project, please check the README.md.

**Comment by pertsevds** [comment]
@r4nchy bro, the answer is simple. It's free!
Want good service? Want simple things? Pay. Use official Sync service.
Or hire vrtmrz to work on this project 24x7.
Or do your sync project.

**Comment by r4nchy** [comment]
@lushangkan the documentation on how to use the plugin effectively is not there. I get nightmare amount of notification from this plugin saying "chunks invalid" etc.
I have already lost few canvas files due to data corruption. Everytime I add notes from one device and then try to access obsidian on another, I have to be on a constant lookout to see if newer notes are getting overwritten by older notes.
And its a nightmare when multiple notes are involved.

There is no documentation on how to use all the features of this plugin without corrupting data and loosing notes.

**Comment by r4nchy** [comment]
@pertsevds you are right, the free stuff is always less quality. The Official Obsidian Sync is also cheaper now. They halved the subscription cost. I was using this plugin as it gives the freedom to host your own server. etc.

**Comment by thany** [comment]
I'm on the fence about this. On the one hand, customisability is good, as it allows software to be useful for a greater amount of people. But the dark side of it is that it makes software look more complicated to novice users.

On the flip side is closed source or commercial software. They often provide little customisation, but this has two reasons. Not only to make their software appear simpler, but most notably because it makes support cheaper. Less options means less things for support staff to worry about.

I feel any software with loads of customisation needs two things:

1. A maintainer who knows what really needs to be customisable, and what doesn't. It's tempting to turn every feature request into an option, but it's important to keep project goals in mind which might include ease of use.
2. Sensible defaults. So a novice user should be able to use the software (somewhat) effectively without _having to_ dive into the customisation options.

To be more specific, I feel obsidian-livesync could do with fewer options, and maybe function more akin to the official Sync mechanism. Maybe it can be smarter in autodetecting what a user otherwise needs to enter manually. I dunno. But safe to say, it _can't_ do without _any_ options other than connection parameters, simply because different people have different needs.

**Comment by vrtmrz** [comment]
I am very sorry for being absent for so long on such an important topic!
First of all, I also would love to express the pleasure, of being open and being in freedom. This includes expressing different perspectives and concerns. Even though we have different perspectives, we will be able to see one future through that. Of course, we have to avoid the "Useless for everyone". So, please do not hesitate, you all. I always appreciate your thoughts and feelings.

Now, back to the topic (Thank you for reading).
Did you tried `Minimal setup` in `Wizard mode`? 
I assume that this is the easiest setup except using set-up uri. And it has been documented as [Quick Setup](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/quick_setup.md). Yes, a short clip is also linked.

If you have not, please read it once and give me feedback! If you have read and felt not enough, please point it out. Of course, I always welcome your contribution too.

**Comment by r4nchy** [comment]
For example I am trying to understand the Plugin sync, as mentioned here
https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/settings.md#enable-plugin-synchronization

but in the extension I don't see anything that says **Plugin Sync** but instead there are **Hidden file sync** and **Customization Sync (Beta)** etc etc.

I mean if there is a easy to follow guide that will have atleast simple setup where we can easily setup multiple devices so that they sync properly with explanation on how to install new plugins or edit configuration on obsidian/plugins so that the new settings can be propagated to new devices without, and how can that be done.

For example I couldn't find any docs on "Customization Sync" like what the `Scan changes", "sync once", "refresh", "Select shiny" , "apply all" button does. 

For example: The QuickAdd Plugin dev has this [video documentation](https://www.youtube.com/watch?v=gYK3VDQsZJo) and I was hoping something like this to exist.

**Comment by ikomhoog** [comment]
Please note that this document you linked is from 2 years ago and starts with:
> NOTE: This document surely became outdated. I'll improve this doc in a while. but your contributions are always welcome.

This is probably outdated like it was already expected to be 2 years ago, so features mentioned have probably been renamed/changed.

>I mean if there is a easy to follow guide that will have atleast simple setup where we can easily setup multiple devices so that they sync properly with explanation on how to install new plugins or edit configuration on obsidian/plugins so that the new settings can be propagated to new devices without, and how can that be done.

There is a simple setup, like vrtmrz already mentioned: https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/quick_setup.md
I followed this and now have a password protected URI that I can just enter into any obsidian client as soon as I install this plugin to set it up.
That's all I need to do to get the syncing going successfully.

Plugins are in a hidden folder ".obsidian" in your vault, due to this it's done through the hidden files part of livesync.
I have hidden files sync enabled and set to overwrite, you choose this with a dialog box when you set it up or first disabling the hidden files sync and then clicking overwrite to enable it again.
Wenever I download a plugin or change a setting a popup on my other device shows up with "click HERE to reload", after that reload the settings have successfully synced.
The only caveat is that it doesn't automatically download the other plugins, instead the plugin shows up in the community plugins -> installed plugins list as disabled.
I assume this is because not all plugins are wanted everywhere or maybe it's a limitation.

I personally turned off the hidden files sync because I didn't want to sync these and use different plugins depending on the device.
I don't need HTML export on my phone for example.

**Comment by r4nchy** [comment]
@vrtmrz my apologies, I was too harsh, I think this is a very well made extension. I was just playing around with a lot of new configuration on the livesync which made the extension unstable as I was using other devices in sync. By looking at the the logs most of the problems are solved.

**Comment by Aetherinox** [comment]
I am going to add in here, because I can see where both sides are coming from.

The addon is feature rich, and it does a great job.

I think what makes it more frustrating is the fact that the documentation is out of date, and another bonus would be having someone assist with the documentation who speaks English natively. That's no fault of the developer, obviously we can't all speak every single language. 

But having a proper updated set of documentation, and written in good English would make things x1000 better. Some of the things I looked for in the documentation aren't there because the docs are outdated. And some things I've had to read about 10 times because the wording is weird, and I'm trying to figure out what the developer means.

Better English docs would make a HUGE difference.

**Comment by r4nchy** [comment]
> some things I've had to read about 10 times because the **wording is weird**, and I'm trying to figure out what the developer means

I think this is so true, the wording is all over the place.
`Hidden File Sync`, `Plugin Sync`, `Customization Sync`, somewhere I saw `Customization Sync` being referred to as just `Custom`

You are right, this is a language barrier (https://github.com/vrtmrz/obsidian-livesync/pull/298). I also think that the documentation should not just explain the words or phrases. But should cover every type of sync usage case by case.

**Comment by Aetherinox** [comment]
Yeah. Like I've thought about forking this repo and going through and re-writing the language translation, but the issue is that I don't even know what the author means with some of the features. So in order to translate it, I would have to physically read through all the code myself to see what each function is doing.

Yes, I know JS, and I work with node daily. But that's a lot of work for someone to do. When it would be much more easily done by someone who has actually done the code and knows what it consists of.

**Comment by gestate-8** [comment]
The reason why official Obsidian Sync's options are simple is that it's functionality is basic.
If you want more functions, naturally, there would be more options.

**Comment by Aetherinox** [comment]
The options are great. The more control I have over something, the better. I like how specific this plugin can be. But with all those features, needs to come good documentation.

And as I said, it's fine that the developer may not speak fluent English. Nobody has to learn every language. But we should try to ensure that thorough documentation is made for the languages available. That way if a user does have a question, they pop open the docs and they've got an answer.

There's quite a few self-hosted solutions I use that are very technical, but the documentation makes it a breeze. 

I've looked through some of the stuff, when the idea of contributing my own translation, but as I said, some of the descriptions are generic, so it's not even fully understood what the feature does unless one of us knows Node / JS and can read through the code and translate that into a simple instruction.

**Comment by gestate-8** [comment]
Yes, of course, the amount of explanation needed will vary from person to person.
Since this is an open-source project, you can directly modify any part that feels inconvenient to you.

**Comment by thany** [comment]
@Aetherinox 

> The options are great. The more control I have over something, the better. I like how specific this plugin can be. But with all those features, needs to come good documentation.

Hard disagree. It doesn't merit a million options just to copy over some files, and retrieve them at the other end. Plus, I would rather have a tool with no options that *just works*, rather than a tool with 17 billion obscure options that requires a prayer and a magical dance to work sometimes.

Without exagerations though, the more options are added to a tool, the more points of failure are introduced. And points of failure Livesync certainly has, ooh boy.

**Comment by ekovarski** [comment]
Your work is greatly appreciated!

We easily forget that the Internet is global and that many other factors come into play such as other cultures and language.

I've returned to LiveSync after using remotely-save because I needed something that was more like Apple Notes in real time. Took me a few hours to wrap my head around it but it works and it's amazing to see me editing a note on one device and seeing the change come across all other devices in near real time. 

Thanks for the plugin, it has some quirks but it's too be expected.

**Comment by aasmoe** [comment]
I like this plugin and it simply works great once it is set up. You just have to hide the things that it puts in the UI. But the user experience is hard. For admin it is fine. But the user experience is so hard that I cant share this service family as I would like to.

Personally I think that for users that are not technically inclined the should be something like:

- Enable the plugin.
- Open plugin settings and be greeted with:
    - Username input
    - Password input
    - Enable/disable livesync
    - [ ] Display advanced settings (unless this is toggled, all settings beyond the above will be hidden)
- Inside Obsidian no sidebar buttons are automatically added (logs, replicate P2P). Only thing is a the status symbol in bottom right.


This is all I really need myself as well. I never really mess with all the settings. 

From reading this thread I understand that this project does not aim to have a simple user experience as the official plugin, which is fair I guess. The sad consequence is that the plugin is overwhelming and difficult to use for many.

For me alone the projects works great.

**Comment by aasmoe** [comment]
Hello @thany. I have only been using it for a few months. I hope I dont get such problems. I find that I can add new clients without too much of a headache. What Im really looking for is something like Sync but self-hosted. I used the official sync for a year or two before. I dont know how this project works with syncing of configurations and such. I wish we had an easy to use self-hosted solution! What type of database would be best suited for syncing?

**Comment by thany** [comment]
> Before you go overboard and declare something you do not understand as 'overkill|', I would really like you to try and imagine why it 's being used. If that is not possible, consider what happens when multiple edits occur simultaneously. I am not asking for everything to be overwritten by the latest date, for instance. 

I've had a myriad of neccesary problems, from files not being synced at all, to files being re-created that had been deleted, and many things in between. Now, I don't know if this is because of obsidian-licesync, or because of CouchDB, as I only have experience with my own situation and don't use CouchDB for anything else. So if you say CouchDB is perfect for multi-client syncing, I believe you, but those problems certainly don't come out of thin air. I would just like you to realise that.

And it's those problems that made it feel unreliable to me. Even if every known issue today is fixed - what's next? That's kind of where my mind goes at that point.

**Comment by vrtmrz** [comment]
I do not want to create an endlessly thorny atmosphere, so let me explain it properly from time to time.

Yes, this is a problem with Self-hosted LiveSync, not CouchDB. I understand this painfully, and I understand very much what you think. Because I use it every day too.

However, if you do not face up to this, it will never improve. I do not think it is very good to say 'this system is not suitable' because it is 'easy to accept and understand, but not the right answer'.
It is the CouchDB mechanism that makes synchronisation possible.

So please read this as an outburst of my sincerity. I am not justifying it, I understand this failure every time.

Of course, I also believe that you cannot trust this mechanism anymore, which is probably the result of honesty as well.
However, I am a little troubled when you give me a wrong explanation for it.
I am sorry if I am a bit wordy, but I am trying to respect you properly and think that you should be the one to point out the problem and understand what is going on.

Let it be condemned as deserved for our faults. But let me make a correction.
You already know that I keep issues opened with `awaiting confirmation` label on, it because I believe that conviction becomes knowledge.

Please refrain from explaining your assumptions as true. 
At least I have done a lot of research before answering you. (The same was true of the CORS case. I also look up old RFCs. I did not explain it to you in my imagination).

Even though this topic is 'Why make it so complex?', and this is something that I, the designer, am accountable for. Hence I do, and hope to reduce misunderstanding the truth. It is a complexed problem. Hence we need some complexed methods. That's all. No silver bullet, and no arrogance. 

I just want to remind you of one thing. We are self-hosting. That means we can all fix this. There are different ways to do it, and this is not limited to source code. If you think the documentation is hanging, that's fine. It can always be better in anyone's hands. I don't think there are any 'customers' or 'consumers' in this.

We all talk positively and hope to make everything better, I hope.

***

Some more explanation: 
Synchronisation is not saving or copying. Nor is it a comparison. It is a tracking. So if you think it is a save or a copy, it will behave completely unexpectedly badly.
For example, many people seem to think that it is easy to synchronise to an 'empty' or 'externally restored initial file', but this is the most difficult kind of synchronisation. It is quite unintuitive. So when we do these things, we need the action of rebuilding. It is not something we do for nothing, but it is very important. And, you may feel so fragile. But it is simply really fragile. Especially (and this is my fault) when this is damaged.

E2EE and chunking come in, which is a real challenge. CouchDB is the solution to this.

Then, if you only have a few devices, use them reasonably one at a time and 'always overwrite with the latest files' is enough for you, then I would not recommend Self-hosted LiveSync always. But if you use five devices sporadically, I can recommend it. It is that kind of plug-in. However, if you have more than one machine or want to synchronise about five thousand files, we want to make sure that this is the best one. 
For reference: when there are more than three devices, 'up-to-date' is actually not easy to define. For example, if you merge old ones with each other, they become new.

**Comment by andreschauer** [comment]
3h to set up Mac→CouchDB→iPhone for a 1.3GB vault. Architecture is solid, onboarding hurts. Three concrete frictions:

Mostly Complete: Decision Required dialog: 3 cryptic options. A first-time user can't tell which one means "upload my vault as canonical". Plain-language labels would fix it.

Doctor errors look like sync failure but aren't: During Mac's initial push, iPhone (secondary device) shows dozens of Failed to gather content of X.md — file-headers arrived before chunks, self-resolves on next pass. To a non-technical user it reads "broken" → instinct is to Discard. Related: #681. Collapse into "Initial pull in progress, X pending" status.

Reverse-proxy gotchas not in docs: Capacitor needs CORS (app://obsidian.md, capacitor://localhost). Traefik default rate-limit (30/min) silently throttles bulk-push to ~6k docs then 429s. A README section "If using Traefik/Caddy/nginx" would save hours.

Constructive ask: would maintainer be open to a community-contributed wrapper-plugin that handles "first 2 devices" with Notion-style UX (passphrase + QR → device-1 auto-pushes → device-2 auto-pulls, no Decision-Required dialogs)? No touch on crypto/sync core, just smoother entry for the 90% case.

Architecture enables best-in-class zero-trust sync. Only onboarding stands between it and 100k+ users.

**Comment by vrtmrz** [comment]
I am sorry for the trouble this has caused you.
Of course, that is fine! Community tools are very welcome.
I have designed, and keep designing the Setup-URI to maintain backwards compatibility.

I agree that the approach you are suggesting is the right one.
In fact, I have recently created a new pane and realised that current dialogues are a bit complicated (finally); I also think onboarding is an issue, so I intend to address it. Basically, as you say, I think the process will involve a flow of Setup-URIs, QR codes or setup, upload and download.

Specifically, I woud like to move Tweaks and Patches to a section called 'System Parameters' before the Remote section, and make it clear that these are settings that need to be included them in the Setup-URI (in other words, new Setup-URI shall be generated) and applied to other devices. Also, even if these are saved to the remote, I woild like to include a configuration timestamp so that they can be automatically identified. 

Besides, now that things are stable, there’s far less need to change these settings. I think it is sufficient to simply provide the default values. Even if the 'System Parameters' section contains quite a lot of explanation, as long as we make it clear that "you don’t usually need to change them", that should not be a problem in itself.

In any case, I am really pleased to hear your suggestion!
This is also a reply to everyone: if you have any suggestions for links to such tools, please do send a pull request to the README.
